install_scripts.sh

```diff
@@ -38,25 +38,27 @@ set -- "${POSITIONAL_ARGS[@]}"
 TOPDIR=$(dirname $(realpath $0))
 
 # Subdirs for generic scripts
-SCRIPTS_DIR_SOURCE=${TOPDIR}/scripts/  # Source dir
-SCRIPTS_DIR_TARGET=${INSTALL_PREFIX}/scripts/  # Target dir
+SCRIPTS_DIR_SOURCE=${TOPDIR}/scripts  # Source dir
+SCRIPTS_DIR_TARGET=${INSTALL_PREFIX}/scripts  # Target dir
 
 # Create target dir
 mkdir -p ${SCRIPTS_DIR_TARGET}
 
 # Copy scripts into this prefix
+echo "copying scripts from ${SCRIPTS_DIR_SOURCE} to ${SCRIPTS_DIR_TARGET}"
 for file in utils.sh; do
-    cp ${SCRIPTS_DIR_SOURCE}/${file} ${SCRIPTS_DIR_TARGET}/${file}
+    cp -v -u ${SCRIPTS_DIR_SOURCE}/${file} ${SCRIPTS_DIR_TARGET}/${file}
 done
 # Subdirs for GPU support
-NVIDIA_GPU_SUPPORT_DIR_SOURCE=${TOPDIR}/gpu_support/nvidia/  # Source dir
-NVIDIA_GPU_SUPPORT_DIR_TARGET=${INSTALL_PREFIX}/gpu_support/nvidia/  # Target dir
+NVIDIA_GPU_SUPPORT_DIR_SOURCE=${TOPDIR}/gpu_support/nvidia  # Source dir
+NVIDIA_GPU_SUPPORT_DIR_TARGET=${INSTALL_PREFIX}/gpu_support/nvidia  # Target dir
```

Comment:
I think if we are shipping scripts then the GPU stuff should appear as a subdir in there, I'll try to make a PR tonight

Reply:
I won't let that block this PR, since we have things in a working state to get `CUDA-Samples` installed, so should be in a follow-up PR.

I'm actually very much in favor of not shipping the scripts in the EESSI repository just yet, and to provide instructions in the documentation to clone the `software-layer` repo instead, for now...

Reply:
You mean `/cvmfs/reponame/scripts/gpu_support`? I thought about that as well. I think it is cleaner, but note that the scripts often assume other files to be present in certain relative paths. So you'll have to make sure to adjust these.

I could try to have a look in a minute...

Reply:
@boegel this PR is already shipping the scripts...

Reply:
For me, I think it's more convenient for the end user to have this self-documenting via Lmod

Reply:
> @boegel this PR is already shipping the scripts...

No, it's not. It's installing the scripts in the build environment, but they are not picked up when creating the tarball for deployment. And even if they would be, the ingestion script running on the Stratum 0 would choke on a tarball that (only) includes stuff in `scripts`.

> For me, I think it's more convenient for the end user to have this self-documenting via Lmod

Why document in two places?
If something goes wrong, the Lmod hook should point to the right page/subsection in the docs (which we need anyway imho).
Also, an Lmod hook that prints a couple of lines and it painful to update isn't the right place to document stuff imho. That's why we created https://eessi.io/docs/gpu as placeholder page.

Reply:
Why make the user go through hoops? Lmod can say:"Run this script"

Reply:
The Lmod hook as to error out anyway, why not have it print useful information

Reply:
And what happens if we change how we do things for a new EESSI version, the Lmod hook would then point to bad jnformation

Reply:
I'm not opposed to providing useful information (or to shipping the scripts in the EESSI repo), I just assume whatever we can print in the Lmod hook won't be *sufficient* for more people.

They'll want to know *what* the script will do exactly, where it will install stuff to, what side effect that has, how to control it, etc.

Reply:
> And what happens if we change how we do things for a new EESSI version, the Lmod hook would then point to bad jnformation

Docs can be updated to have version-specific info, so that's not exactly true.
Lmod hook could be updated to point to `https://eessi.io/docs/gpu/2027.09`, and `https://eessi.io/docs/gpu` could be set up to auto-redirect to `https://eessi.io/docs/gpu/2023.06`.

Reply:
You don't think thats a lot of extra effort for little gain?

Personally, I think there are plenty  of people who do not care _how_ it works, just that _it works_.

Reply:
You can always add the docs pointers in the scripts if they're actually needed

Reply:
For some people, just getting it to work will be enough.

Not for system administrators though, they won't go and run a random script just because they're told to.
They may even want to do what needs to be done some other way (through Puppet or whatever).

Reply:
Sure but then you are sculpting things for minority use case (sysadmins).

Anyway, we can do both, the Lmod hook can mention the script and the doc, let the users decide